pkg/network/openshift_sdn.go

```diff
@@ -266,3 +286,20 @@ func clusterNetwork(conf *operv1.NetworkSpec) (string, error) {
 
 	return string(cnBuf), nil
 }
+
+func bootstrapSDN(conf *operv1.Network, kubeClient client.Client) (*bootstrap.BootstrapResult, error) {
+
+	infraConfig := &configv1.Infrastructure{}
+	if err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
+		return nil, fmt.Errorf("failed to get infrastructure 'config': %v", err)
+	}
+
+	klog.V(2).Infof("Openshift-SDN: Bootstrap SDN infraConfig Platform: %v ", infraConfig.Status.PlatformStatus.Type)
```

Comment:
Check for nil `infraConfig.Status.PlatformStatus` (it could be nil if the cluster was installed with OpenShift 4.1 and upgraded).

Reply:
ack s/nil/""/

Reply:
`infraConfig.Status.PlatformStatus` itself could be nil.  The API field was added in OpenShift 4.2, and a cluster that was installed on OpenShift 4.1 and upgraded 4.1 → 4.2 → ... → 4.8 might not have this field set.  We've had BZs in the past when we forgot to check `Status.PlatformStatus` for nil (e.g., https://bugzilla.redhat.com/show_bug.cgi?id=1734192, https://bugzilla.redhat.com/show_bug.cgi?id=1834194).  We should try not to repeat that mistake yet another time.  * grin *.